System/DataModule.cpp

```diff
@@ -1,764 +1,497 @@
-//////////////////////////////////////////////////////////////////////////////////////////
-// File:            DataModule.cpp
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Source file for the DataModule class.
-// Project:         Retro Terrain Engine
-// Author(s):       Daniel Tabar
-//                  data@datarealms.com
-//                  http://www.datarealms.com
+#include "DataModule.h"
+#include "RTEManagers.h"
 
+namespace RTE {
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Inclusions of header files
+	const std::string DataModule::m_ClassName = "DataModule";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void DataModule::Clear() {
+		m_FileName.clear();
+		m_FriendlyName.clear();
+		m_Author.clear();
+		m_Description.clear();
+		m_Version = 1;
+		m_ModuleID = -1;
+		m_IconFile.Reset();
+		m_pIcon = 0;
+		m_PresetList.clear();
+		m_EntityList.clear();
+		m_TypeMap.clear();
+		for (int i = 0; i < c_PaletteEntriesNumber; ++i) { m_MaterialMappings[i] = 0; }
+
+		m_ScanFolderContents = false;
+		m_IgnoreMissingItems = false;
+		m_CrabToHumanSpawnRatio = 0;
+		m_ScriptPath.clear();
+	}
 
-#include "DataModule.h"
-#include "RTEManagers.h"
-#include "Entity.h"
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#include "allegro.h"
+	int DataModule::Create(std::string moduleName, ProgressCallback fpProgressCallback) {
+		m_FileName = moduleName;
+		m_ModuleID = g_PresetMan.GetModuleID(moduleName);
+		m_CrabToHumanSpawnRatio = 0;
 
-namespace RTE {
+		// Report that we're starting to read a new DataModule
+		if (fpProgressCallback) {
+			char report[512];
+			sprintf_s(report, sizeof(report), "%s %c loading:", m_FileName.c_str(), -43);
+			fpProgressCallback(std::string(report), true);
+		}
+
+		Reader reader;
+		std::string indexPath(m_FileName + "/Index.ini");
+		std::string mergedIndexPath(m_FileName + "/MergedIndex.ini");
+
+		// Look for merged version of the index file, a single file where all the INI data was merged to load faster
+		if (std::experimental::filesystem::exists(mergedIndexPath.c_str())) { indexPath = mergedIndexPath; }
```

Comment:
This is about the time we experimented with speeding up the loading times before coming to the conclusion the actual drawing of the progress report was slowing us down.

This looks for a `MergedIndex.ini` in a module and loads it instead of `Index.ini`
MergedIndex is created with Weegee's merger tool, it just takes all the ini in the whole module and consolidates them into one file in the order of appearance, while removing any commented code and most (or all, don't remember) of the whitespace.

This proved some minor improvement in loading times but eventually just disabling the drawing is the big improvement.

I think it's still a nice thing to have and we should keep it, because it still speeds things up by not executing all the extra file search, open, read etc etc if the mod author merged the index.
It's probably in the microseconds, but gainzz are gainzz.

Reply:
Okay, that's what I though.
Sure, let's keep it and put the merger tool in data. I think eventually this'll be pointless as we find more and better ways to speed up loading, but it's not a lot of code cruft so why not. Would you mind you expand the comment to mention that the merged index is made by the merger tool, so it's 100% clear what this is for?